pkg/sessionctx/variable/session.go

```diff
@@ -1514,6 +1514,11 @@ type SessionVars struct {
 	// use the ExpectedCnt to adjust the estimated row count for index scan.
 	OptOrderingIdxSelThresh float64
 
+	// OptOrderingIdxSelRatio is the ratio for optimizer to determine when qualified rows from filtering outside
+	// of the index will be found during the scan of an ordering index.
+	// If all filtering is applied as matching on the ordering index, this ratio will have no impact.
+	OptOrderingIdxSelRatio float64
```

Comment:
It seems that every query need different OptOrderingIdxSelRatio, even th values of different "where column=value" in the same query template should use different configurations.
So I feel it should be more like a hint rather than a session variable

Reply:
This was modeled after the prior solution for OptOrderingIdxSelThreshold - which has the same challenge. However, a hint is not the preferred approach here - since we already have a way to hint an individual query with hinting the individual index. We need a solution that applies more generally - which is why this has a global variable.